### PR TITLE
RMB-849: Remove ModuleName class from domain-models-api-interfaces

### PR DESCRIPTION
**Pull request by julianladisch**
domain-models-api-interfaces (and domain-models-runtime
that depends on it) ships with a ModuleName class with
getModuleName() returning "raml_module_builder".

An RMB based module creates a correct ModuleName class
when running domain-models-maven-plugin.

If we are lucky the latter is included in the final jar.
For mod-feesfines I sometimes have integration tests failure
because of wrong order resulting in the
wrong "raml_module_builder" where it should be "mod_feesfines".

Removing ModuleName class from domain-models-api-interfaces jar
resolves this issue; it requires reflection to load
ModuleName class, though.

### CODE REVIEW

File: domain-models-runtime/src/test/java/org/folio/rest/tools/utils/ModuleName.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package org.folio.rest.tools.utils;
+
+public class ModuleName {
+  /**
+   * The module name with minus replaced by underscore, for example {@code mod_foo_bar}.
+   */
+  public static String getModuleName() {
```
Comment (adamdickmeiss) [inline domain-models-runtime/src/test/java/org/folio/rest/tools/utils/ModuleName.java:7]: I don't like that this class and method is bundled with the runtime.. Can't this just be in test?

Comment (julianladisch) [reply domain-models-runtime/src/test/java/org/folio/rest/tools/utils/ModuleName.java:7]: For me `domain-models-runtime/src/test` is in test only.
Can you elaborate?

Comment (adamdickmeiss) [reply domain-models-runtime/src/test/java/org/folio/rest/tools/utils/ModuleName.java:7]: Brain damage on my side.. Excellent.

**Review by adamdickmeiss — APPROVED** [review]
LGTM. Should be tested with one real module before merging (if not done already).

File: domain-models-runtime/src/test/java/org/folio/rest/tools/utils/ModuleName.java (hunk 1 above)
Comment (julianladisch) [reply domain-models-runtime/src/test/java/org/folio/rest/tools/utils/ModuleName.java:7]: ModuleName is not in the fat jar:
```
unzip -l domain-models-runtime/target/domain-models-runtime-fat.jar|grep -i ModuleName
```

**Comment by julianladisch** [comment]
Successfully tested on mod-inventory-storage, mod-feesfines and mod-login-saml.

**Pull request merged**